sdk/search/Azure.Search.Documents/api/Azure.Search.Documents.netstandard2.0.cs

```diff
@@ -212,9 +212,9 @@ public partial class AnalyzedTokenInfo
         public int StartOffset { get { throw null; } }
         public string Token { get { throw null; } }
     }
-    public partial class AnalyzeRequest
+    public partial class AnalyzeTextRequest
```

Comment:
Why not AnalyzeTextOptions?  I thought that's where we landed in the meeting.

Reply:
See my thread in our Teams channel. Basically, nothing about it is optional. According to [guidelines](https://azure.github.io/azure-sdk/general_design.html#client-interface):
> There are times when operations require the addition of optional data, provided in what is colloquially known as an “options bag”. Libraries should strive for consistent naming.
But `Text` is not optional, and one of `Analyzer` or `Tokenizer` is required by the service. The other option is, then, to have overloads requiring one or the other, but:
1. Not all languages can do that, and the type (string) would be the same anyway, and
2. That sounds like client-side validation of non-path arguments, which is against [guidelines](https://azure.github.io/azure-sdk/general_implementation.html#parameter-validation).
> DO NOT validate service parameters. This includes null checks, empty strings, and other common validating conditions. Let the service validate any request parameters.
If you or @KrzysztofCwalina have a different interpretation or a better suggestion, I'm happy to make a change.

Reply:
/cc @sima-zhu @alzimmermsft @xiangyan99 @sarangan12 @xirzec

Reply:
Here's the C# specific guidance:  https://azure.github.io/azure-sdk/dotnet_introduction.html#dotnet-parameters  (The example code has a bug where `BlobCreateOptions` doesn't have `string BlobName` but Krzysztof is going to fix that.)

We had a meeting earlier this week to clarify it a bit in relation to Java.  My current understanding is that `*Options` doesn't mean optional.  Here's what I'd expect now:

```C#
AnalyzeText(string text, Analyzer analyzer);
AnalyzeText(string text, Tokenizer tokenizer);
AnalyzeText(AnalyzeTextOptions options);

public class AnalyzeTextOptions
{
    public AnalyzeTextOptions(string text, Analyzer analyzer);
    public AnalyzeTextOptions(string text, Tokenizer tokenizer);
    public string Text { get; }
    public Analyzer Analyzer { get; }
    public Tokenizer Tokenizer { get; }
    public object WhateverElseMutable { get; set; }
}
```

I think it's too late to change this consistently everywhere, but I see no harm in calling the type `AnalyzeTextOptions`.  I don't feel strongly though so I'll defer to Krzysztof.

Reply:
Java has made the change to AnalyzeTextRequest. 
@JonathanGiles could you suggest the best naming for Java?

Reply:
[Guidance ](https://azure.github.io/azure-sdk/java_design.html#model-types)for java

Reply:
Java is a bit looser in their description:
> Optional parameters and settings to an operation should be collected into an options bag named <operation>Options

As such, I'm fine renaming this to `AnalyzeTextOptions` but not the overloads, which is too much like client-side validation of model properties. The swagger lists only `text` as required, though it probably should've used `oneOf` for the others (though, I'm not sure generators support that - just `required`). We can always add it later if we see a lot of confusion by customers.

Reply:
https://github.com/Azure/azure-sdk-for-java/pull/11783

Reply:
"Options" in the context of the option bag parameter names does not mean "optional". It's more like options on the fix prix menu, where you need to pick at least one option or you will end up hungry, yet with ah check to pay.

Reply:
@KrzysztofCwalina should we also have overloads (either method and/or model constructor overloads) that take either an `Analyzer` or `Tokenizer`? That really trends toward client-side validation of properties that aren't listed in swagger's `required`, but could be helpful. If so, I can get bugs opened across languages immediately.

Reply:
I'm going to send out an email to clarify our intent and open bugs across languages.